### PR TITLE
HPCGAP: Make product of two lists behave as in GAP

### PR DESCRIPTION
**Pull request by ssiccha**
#1133 
### Tick all what applies to this pull request
- [ ] Adds new features
- [X] Improves and extends functionality
- [ ] Fixes bugs that could lead to crashes
- [ ] Fixes bugs that could lead to incorrect results
- [ ] Fixes bugs that could lead to break loops

### Write below the description of changes (for the release notes)
The product of two lists L and H returns the empty list, if L is empty.

### CODE REVIEW

File: lib/list.gi [hunk 1]
```diff
@@ -3371,7 +3371,11 @@ InstallOtherMethod( \*,
           if depthDiff < -1 or depthDiff > 1 then
               return PROD_LISTS_SPECIAL(left, right, depth1, depth2 );
           else
-              return PROD_LIST_LIST_DEFAULT( left, right, depthDiff );
+              if IsEmpty( left ) then
+                  return [];
```
Comment (markuspf) [inline lib/list.gi]: What happens if `right` is empty?

**Comment by markuspf** [comment]
This PR needs tests.

File: lib/list.gi (hunk 1 above)
Comment (fingolfin) [inline lib/list.gi]: Also, does this now match the corresponding code in master (I am AFK so can't check right now)

Comment (ssiccha) [inline lib/list.gi]: I kind of expected it to fail, but missed to check that. GAP throws an error when the right list is empty.
```
gap> l := [1,2,3];
[ 1, 2, 3 ]
gap> l * [];
Error, Inner product multiplication of lists: no summands in
  return PROD_LIST_LIST_DEFAULT( left, right, depthDiff )
 ; at /opt/gap/gap4r8/lib/list.gi:3250 called from 
<function "unknown">( <arguments> )
 called from read-eval loop at line 2 of *stdin*
```

Comment (ssiccha) [inline lib/list.gi]: Now, should GAP be able to do both or none? It feels inconsistent to allow multiplication with an empty list from the left, but not from the right.

Comment (ssiccha) [inline lib/list.gi]: @fingolfin I also just thought about that. I will make sure that it will not conflict with the master-branch.

**Comment by ssiccha** [comment]
I will have to look at the manual to add some tests.

File: lib/list.gi (hunk 1 above)
Comment (fingolfin) [inline lib/list.gi]: I just checked, and in the master branch, this function is identical. So this seems like a cure for the symptom, not the actual cause.

**Comment by markuspf** [comment]
If I try this in GAP I get
```
gap> [1,2,3] * [ ];
Error, Inner product multiplication of lists: no summands in
  return PROD_LIST_LIST_DEFAULT( left, right, depthDiff ); at /home/makx/ac/gap/lib/list.gi:3397 called from 
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:1
brk> 
gap> [ ] * [1,2,3];
Error, Inner product multiplication of lists: no summands in
  return PROD_LIST_LIST_DEFAULT( left, right, depthDiff ); at /home/makx/ac/gap/lib/list.gi:3397 called from 
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:1
brk> 
```

Do you have any packages loaded that change the behaviour of `*` on lists?

**Comment by ssiccha** [comment]
Ah, the code I was working with loads the Gauss package. I'm afk at the moment. I will check when I'm back.

**Comment by ssiccha** [comment]
Ah, I just realized my error. The code behaving differently is actually
```
gap> [ [] ] * [ [1,2,3] ];
[ [  ] ]
```
which in HPCGAP throws the error
```
Error, Inner product multiplication of lists: no summands in
   called from 
return PROD_LIST_LIST_DEFAULT( left, right, depthDiff )
 ; at /home/sergio/hpcgap/gap/lib/list.gi:3374PROD_LIST_SCL_DEFAULT( left, right 
 ) at /home/sergio/hpcgap/gap/lib/list.gi:3393 called from
<function "unknown">( <arguments> )
 called from read-eval loop at line 1 of stream
```
Something else must be off in HPCGAP. I will close this pull request and write more in the actual issue.

**Pull request closed without merging**